gdal/frmts/esric/esric_dataset.cpp

```diff
@@ -0,0 +1,446 @@
+    nBlockXSize = nBlockYSize = 256;
+
+    // Default color interpretation
+    ci = (parent->nBands > 2) ? rgba[b - 1] : la[b - 1];
+    if (0 == lvl)
+        AddOverviews();
+}
+
+void ECBand::AddOverviews() {
+    auto parent = reinterpret_cast<ECDataset*>(poDS);
+    for (size_t i = 1; i < parent->resolutions.size(); i++) {
+        ECBand* ovl = new ECBand(parent, nBand, int(i));
+        if (!ovl)
+            break;
+        overviews.push_back(ovl);
+    }
+}
+
+CPLErr ECBand::MissingTile(int nBlockXOff, int nBlockYOff, void* pData) {
+    auto parent = reinterpret_cast<ECDataset*>(poDS);
+    auto& buffer = parent->tilebuffer;
+    size_t TSZ = parent->TSZ;
+    if (1 == nBand) {
+        buffer.resize(TSZ * TSZ * parent->nBands);
+        fill(buffer.begin(), buffer.end(), static_cast<GByte>(0));
+        for (int iBand = 2; iBand <= parent->nBands; iBand++) {
+            auto band = parent->GetRasterBand(iBand);
+            if (lvl)
+                band = band->GetOverview(lvl - 1);
+            GDALRasterBlock* const poBlock = band->GetLockedBlockRef(nBlockXOff, nBlockYOff);
+            if (poBlock != nullptr)
+                poBlock->DropLock();
+        }
+    }
+    memset(pData, 0, TSZ * TSZ);
+    return CE_None;
+}
+
+CPLErr ECBand::IReadBlock(int nBlockXOff, int nBlockYOff, void* pData) {
+    auto parent = reinterpret_cast<ECDataset*>(poDS);
+    auto& buffer = parent->tilebuffer;
+    auto const TSZ = parent->TSZ;
+    auto const BSZ = parent->BSZ;
+
+    if (1 == nBand) {
+        buffer.resize(size_t(TSZ) * TSZ * parent->nBands);
+
+        int lxx = static_cast<int>(parent->resolutions.size() - lvl - 1);
+        int bx, by;
+        bx = (nBlockXOff / BSZ) * BSZ;
+        by = (nBlockXOff / BSZ) * BSZ;
+        CPLString fname;
+        fname = CPLString().Printf("%s/L%02d/R%04xC%04x.bundle", parent->dname.c_str(), lxx, by, bx);
+        Bundle& bundle = parent->GetBundle(fname);
+        if (nullptr == bundle.fh) { // This is not an error in general, bundles can be missing
+            CPLError(CE_Debug, CPLE_FileIO, "Can't open bundle %s", fname.c_str());
+            return MissingTile(nBlockXOff, nBlockYOff, pData);
+        }
+        int block = static_cast<int>((nBlockYOff % BSZ) * BSZ + (nBlockXOff % BSZ));
+        GUInt64 offset = bundle.index[block] & 0xffffffffffull;
+        GUInt64 size = bundle.index[block] >> 40;
+        if (0 == size)
+            return MissingTile(nBlockXOff, nBlockYOff, pData);
+        auto& fbuffer = parent->filebuffer;
+        fbuffer.resize(size_t(size));
+        VSIFSeekL(bundle.fh, offset, SEEK_SET);
+        if (size != VSIFReadL(fbuffer.data(), size_t(1), size_t(size), bundle.fh)) {
+            CPLError(CE_Failure, CPLE_FileIO, "Error reading tile");
+            return CE_Failure;
+        }
+        CPLString magic;
+        // Should use some sort of unique
+        magic.Printf("/vsimem/esric_%u.tmp", rand());
```

Comment:
As I understand the code, the life time of this temporary in-memory file is just this method, so the following should do:
```suggestion
        magic.Printf("/vsimem/esric_%p.tmp", this);
```
and remove above comment

Reply:
makes sense, thanks